### DISCUSSION TITLE
[@types/minipass]: error TS2416: Property 'end' in type 'MiniPass' is not assignable to the same property in base type 'WritableStream'.

### DISCUSSION THREAD

**Discussion by akashg3627** [DefinitelyTyped/DefinitelyTyped#58771] [Issues with a @types package]
Hello, I have come across this issue with the latest version `@types/node@17.0.17 `

` 
node_modules/@types/minipass/index.d.ts:32:5 - error TS2416: Property 'end' in type 'MiniPass' is not assignable to the same property in base type 'WritableStream'.Type '{ (cb?: () => void): void; (chunk: any, cb?: () => void): void; (chunk: any, encoding?: string, cb?: () => void): void; }' is not assignable to type '{ (cb?: () => void): this; (data: string | Uint8Array, cb?: () => void): this; (str: string, encoding?: BufferEncoding, cb?: () => void): this; }'.end(chunk: any, encoding?: string | null, cb?: () => void): void;
` 

After going through some past discussions, I found that this issue was resolved in some earlier versions but has come up again.
Past discussion #57970
Past Issue #57994
Any quick help will be highly appreciated.
Thank you

[a comment by an automated account is not shown]

**Comment by PaulRBerg** [comment]
Yeah can confirm that this issue is still occurring. I'm using node v16.14.0, @types/node v17.0.21 and typescript v.4.6.2.

```text
error TS2416: Property 'end' in type 'MiniPass' is not assignable to the same property in base type 'WritableStream'.
  Type '{ (cb?: (() => void) | undefined): void; (chunk: any, cb?: (() => void) | undefined): void; (chunk: any, encoding?: string | null | undefined, cb?: (() => void) | undefined): void; }' is not assignable to type '{ (cb?: (() => void) | undefined): this; (data: string | Uint8Array, cb?: (() => void) | undefined): this; (str: string, encoding?: BufferEncoding | undefined, cb?: (() => void) | undefined): this; }'.
    Type 'void' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'void'.

27     end(cb?: () => void): void;
```
